crates/rerun/src/run.rs

```diff
@@ -280,6 +303,60 @@ where
     }
 }
 
+/// Checks whether two .rrd files are _similar_, i.e. not equal on a byte-level but
+/// functionally equivalent.
+///
+/// Returns `Ok(())` if they match, or an error containing a detailed diff otherwise.
+fn run_compare(path_to_rrd1: &Path, path_to_rrd2: &Path, full_dump: bool) -> anyhow::Result<()> {
+    /// Given a path to an rrd file, builds up a `DataStore` and returns its contents as one big
+    /// `DataTable`.
+    fn compute_uber_table(path_to_rrd: &Path) -> anyhow::Result<re_log_types::DataTable> {
+        let rrd = {
+            let bytes = std::fs::read(path_to_rrd)
+                .with_context(|| format!("couldn't read rrd file contents at {path_to_rrd:?}"))?;
+            re_log_encoding::decoder::decode_bytes(&bytes)?
+        };
+
+        let store_id = {
+            let store_info = rrd
+                .iter()
+                .find(|msg| matches!(msg, LogMsg::SetStoreInfo(_)))
+                .ok_or_else(|| {
+                    anyhow::anyhow!("couldn't find store ID for rrd file at {path_to_rrd:?}")
+                })?;
+            store_info.store_id().clone()
+        };
+
+        let mut store = re_data_store::StoreDb::new(store_id);
+        for msg in &rrd {
+            store.add(msg)?;
+        }
```

Comment:
This fails silently if there are multiple record-ids in the same .rrd file. You should use `StoreBundle::from_rrd` instead

Reply:
Ha, nice! It sure felt weird having to do this